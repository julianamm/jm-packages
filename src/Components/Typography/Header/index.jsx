import React from 'react';
import Proptypes from 'prop-types';
import {
    H1,
    H2,
    H3,
    H4,
    H5,
    H6,
} from './styles';

const Header = props => {

    switch ( true ) {
        case props.h1 :
            return <H1 {...props}>{props.label}</H1>
        case props.h3 :
            return <H3 {...props}>{props.label}</H3>
        case props.h4 :
            return <H4 {...props}>{props.label}</H4>
        case props.h5 :
            return <H5 {...props}>{props.label}</H5>
        case props.h6 :
            return <H6 {...props}>{props.label}</H6>

        default:
            return <H2 {...props} >{props.label}</H2>
    }
}

Header.defaultProps = {
    h1: false,
    h2: true,
    h3: false,
    h4: false,
    h5: false,
    h6: false,
    color: 'Blue-Grey-900',
    hover: false,
    textCenter: false,
}

Header.propTypes = {
    label: Proptypes.string.isRequired,
    h1: Proptypes.bool,
    h2: Proptypes.bool,
    h3: Proptypes.bool,
    h4: Proptypes.bool,
    h5: Proptypes.bool,
    h6: Proptypes.bool,
    color: Proptypes.string,
    hover: Proptypes.bool,
    textCenter: Proptypes.bool,
}

export default Header;