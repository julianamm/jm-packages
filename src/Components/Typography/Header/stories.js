import React from 'react'
import { storiesOf } from '@storybook/react'
import Header from '.'
import Body from '../Body'

storiesOf('Typography/Headers', module)

.add('Header e Body', () => (
  <>
    <Header label={"Header Default"} />
    <Body label={"Body label "} />
  </>
))

.add('Default', () => (
  <>
    <Header label={"Header Default"} />
    <Header h1 label={"Header H1"} />
    <Header h2 label={"Header H2"} />
    <Header h3 label={"Header H3"} />
    <Header h4 label={"Header H4"} />
    <Header h5 label={"Header H5"} />
    <Header h6 label={"Header H6"} />
  </>
))

.add('Colorful', () => (
  <>
    <Header label={"Header Default"} />
    <Header color="Deep-Purple-A200" h1 label={"Header H1 Deep-Purple-A200"} />
    <Header color={"Light-Blue-100"} h2 label={"Header H2 Light-Blue-100"} />
    <Header color={"Teal-400"} h3 label={"Header H3 Teal-400"} />
    <Header color={"Pink-A200"} h4 label={"Header H4 Pink-A200"} />
    <Header color={"Lime-A400"} h5 label={"Header H5 Lime-A400"} />
    <Header color={"Blue-A200"} h6 label={"Header H6 Blue-A200"} />
  </>
))