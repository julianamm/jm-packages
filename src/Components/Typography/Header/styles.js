import styled from 'styled-components'
import Color from '../../Colors'

export const H1 = styled.h1`
    margin: 0;
    padding: 0;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

    ${({ textCenter }) => textCenter && `
        text-align: center;
    `}

    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}
`

export const H2 = styled.h2`
    margin: 0;
    padding: 0;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

    ${({ textCenter }) => textCenter && `
        text-align: center;
    `}

    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}
`

export const H3 = styled.h3`
    margin: 0;
    padding: 0;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

    ${({ textCenter }) => textCenter && `
        text-align: center;
    `}

    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}
`

export const H4 = styled.h4`
    margin: 0;
    padding: 0;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

    ${({ textCenter }) => textCenter && `
        text-align: center;
    `}

    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}
`

export const H5 = styled.h5`
    margin: 0;
    padding: 0;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

    ${({ textCenter }) => textCenter && `
        text-align: center;
    `}

    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}
`

export const H6 = styled.h6`
    margin: 0;
    padding: 0;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

        ${({ textCenter }) => textCenter && `
        text-align: center;
    `}

    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}
`
