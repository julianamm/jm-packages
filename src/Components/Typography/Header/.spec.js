import React from 'react'
import renderer from 'react-test-renderer'

import Header from '.'

describe('Typography', () => {
    it('Rendering Font Components without crashing', () => {
        const tree = renderer.create(
            <Header
                color={"Indigo-900"}
                h1
                label={"Header component Indigo-900"}
            />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})