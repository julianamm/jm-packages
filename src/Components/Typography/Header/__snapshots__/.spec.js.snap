// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendering Font Components without crashing 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #1A237E;
}

<h1
  className="c0"
  color="Indigo-900"
  label="Header component Indigo-900"
>
  Header component Indigo-900
</h1>
`;
